internal/servers/controller/handlers/sessions/session_service.go

```diff
@@ -120,19 +124,25 @@ func (s Service) ListSessions(ctx context.Context, req *pbs.ListSessionsRequest)
 		item.Scope = scopeInfoMap[item.GetScopeId()]
 		res.ScopeId = item.Scope.Id
 		authorizedActions := authResults.FetchActionSetForId(ctx, item.Id, IdActions, auth.WithResource(res))
-		if len(authorizedActions) > 0 {
-			onlySelf := true
-			for _, v := range authorizedActions {
-				if v != action.ReadSelf && v != action.CancelSelf {
-					onlySelf = false
-					break
-				}
-			}
-			if !onlySelf || item.GetUserId() == authResults.UserId {
-				item.AuthorizedActions = authorizedActions.Strings()
-				finalItems = append(finalItems, item)
+		if len(authorizedActions) == 0 {
+			continue
+		}
+		onlySelf := true
+		for _, v := range authorizedActions {
+			if v != action.ReadSelf && v != action.CancelSelf {
+				onlySelf = false
+				break
 			}
 		}
+		if onlySelf && item.GetUserId() == authResults.UserId {
+			continue
```

Comment:
I don't think this logic is right...if you're going to flip the logic from what it was before, you need `!=` here.

The item.AuthorizedActions line below should probably also go in the `if filter.Match` clause too.

Reply:
Woops, thanks for catching that! I'll update it any place i missed up the flipping of the logic.

Re: putting item.AuthorizedActions inside the filter.Match clause, if we perform this outside of the filter.Match clause we allow the filter to use the `authorized_actions` as part of the filter.